src/gt4py/next/iterator/transforms/constant_folding.py

```diff
@@ -20,38 +20,129 @@ def visit_FunCall(self, node: ir.FunCall):
+        if cpm.is_call_to(new_node, ("minimum", "maximum")):
+            if all(cpm.is_call_to(arg, "maximum") for arg in new_node.args) or all(
+                cpm.is_call_to(arg, "minimum") for arg in new_node.args
+            ):
+                if (
+                    new_node.args[0].args[0] == new_node.args[1].args[1]
+                    and new_node.args[0].args[1] == new_node.args[1].args[0]
+                ):
+                    new_node = new_node.args[0]
+        # `maximum(maximum(__out_size_1, 1), __out_size_1)` -> `maximum(__out_size_1, 1)`
+        if cpm.is_call_to(new_node, ("minimum", "maximum")):
+            match = False
+            if isinstance(new_node.args[0], ir.FunCall) and isinstance(
+                new_node.args[1], (ir.Literal, ir.SymRef)
+            ):
+                fun_call, sym_lit = new_node.args
+                match = True
+            elif isinstance(new_node.args[0], (ir.Literal, ir.SymRef)) and isinstance(
+                new_node.args[1], ir.FunCall
+            ):
+                match = True
+                sym_lit, fun_call = new_node.args
+            if match and cpm.is_call_to(fun_call, ("maximum", "minimum")):
+                if isinstance(fun_call.args[0], ir.SymRef) and isinstance(
+                    fun_call.args[1], ir.Literal
+                ):
+                    if sym_lit == fun_call.args[0]:
+                        new_node = im.call(fun_call.fun.id)(sym_lit, fun_call.args[1])
+                    if sym_lit == fun_call.args[1]:
+                        new_node = im.call(fun_call.fun.id)(fun_call.args[0], sym_lit)
+                if isinstance(fun_call.args[0], ir.Literal) and isinstance(
+                    fun_call.args[1], ir.SymRef
+                ):
+                    if sym_lit == fun_call.args[0]:
+                        new_node = im.call(fun_call.fun.id)(fun_call.args[1], sym_lit)
+                    if sym_lit == fun_call.args[1]:
+                        new_node = im.call(fun_call.fun.id)(sym_lit, fun_call.args[0])
+        # `maximum(plus(__out_size_1, 1), minus(__out_size_1, 1))` -> `plus(__out_size_1, 1)`
+        if cpm.is_call_to(new_node, ("minimum", "maximum")):
+            if all(cpm.is_call_to(arg, ("plus", "minus")) for arg in new_node.args):
+                if new_node.args[0].args[0] == new_node.args[1].args[0]:
+                    new_node = im.plus(
+                        new_node.args[0].args[0],
+                        ConstantFolding.apply(
+                            im.call(new_node.fun.id)(
+                                im.call(new_node.args[0].fun.id)(0, new_node.args[0].args[1]),
+                                im.call(new_node.args[1].fun.id)(0, new_node.args[1].args[1]),
+                            )
+                        ),
+                    )
+            # `maximum(plus(__out_size_1, 1), __out_size_1)` -> `plus(__out_size_1, 1)`
```

Comment:
Repetition of the case somewhere above right? I think we have reached the point where the structure of this pass does not work well anymore since previous transformations might allow other transformations to run again. The CollapseTuple pass already has some functionality to solve this where transformations are executed until a fixed point is reached (`fp_transform` and `transform`). We can take a look together.